301d58/code-challenges/map/challenges-07.test.js

```diff
@@ -0,0 +1,277 @@
+
+const mapTwoToThe = (arr) => {
+  let mapArr = arr.map( function(value){
+    return (Math.pow(2,value));
+  });
+  return mapArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named charCode that, given an array of letters as an input, uses map to return a new array where each element
+is the result of the `charCodeAt` method on the original array element.
+
+Read the MDN documentation on String.charCodeAt() if necessary.
+
+For example: charCode(['h','i']) returns [104, 105].
+------------------------------------------------------------------------------------------------ */
+
+const charCode = (arr) => {
+  let newArr = arr.map( function(value) {
+    return value.charCodeAt(0);
+  });
+  return newArr
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function that, given an array of numbers as input, uses map to return a new array where each element is either the string
+"even" or the string "odd", based on each value.
+
+If any element in the array is not a number, the resulting array should have the string "N/A" in its place.
+
+For example: evenOdd([1,2,3]) returns ['odd','even','odd'].
+------------------------------------------------------------------------------------------------ */
+
+const evenOdd = (arr) => {
+    let newArr =  arr.map( function (value) {
+    if (typeof value !== 'number') {
+      return 'N/A';
+    }
+    if (value % 2 === 0){
+      return 'even';
+    } else {
+      return 'odd';
+    }
+  });
+  return newArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+Use the snorlaxAbilities data, below, for this challenge.
+
+Write a function named extractAbilities that, given the array of abilities, uses map to create an array containing only the ability name.
+
+Note: Because this function is expecting the array of abilities, it will be invoked as:
+extractAbilities(snorlaxAbilities.abilities)
+------------------------------------------------------------------------------------------------ */
+
+const snorlaxAbilities = {
+  abilities: [
+    {
+      slot: 3,
+      is_hidden: true,
+      ability: {
+        url: 'https://pokeapi.co/api/v2/ability/82/',
+        name: 'gluttony',
+      },
+    },
+    {
+      slot: 2,
+      is_hidden: false,
+      ability: {
+        url: 'https://pokeapi.co/api/v2/ability/56/',
+        name: 'cute charm',
+      },
+    },
+    {
+      slot: 1,
+      is_hidden: false,
+      ability: {
+        url: 'https://pokeapi.co/api/v2/ability/17/',
+        name: 'immunity',
+      },
+    },
+  ],
+  name: 'snorlax',
+  weight: 4600,
+};
+
+const extractAbilities = (arr) => {
+   let newArr = arr.map( function (value) {value.ability.name});
```

Comment:
I think this isn't working because your callback is a 'regular' function and doesn't have a return statement. If this was an arrow function, you could put the entire .map() on one line and it would return because arrow function one liners have implicit return.